xfields/beam_elements/beambeam_src/beambeam3d.h

```diff
@@ -109,6 +125,35 @@ void synchrobeam_kick(
     double Fz_star = 0.5*(Fx_hat_star*dS_x_bar_hat_star  + Fy_hat_star*dS_y_bar_hat_star+
                    Gx_hat_star*dS_Sig_11_hat_star + Gy_hat_star*dS_Sig_33_hat_star);
 
+/*
+    printf("\tEx=%.10e\n", Ex);
+    printf("\tEy=%.10e\n", Ey);
+    printf("\tGx=%.10e\n", Gx);
+    printf("\tGy=%.10e\n", Gy);
+    printf("\tFx_boosted=%.10e\n", Fx_star);
+    printf("\tFy_boosted=%.10e\n", Fy_star);
+    printf("\tFz_boosted=%.10e\n", Fz_star);
+    printf("\trpp=%.20e\n", LocalParticle_get_rpp(part));
+*/
+
+    // emit beamstrahlung photons from single macropart
+    if (do_beamstrahlung==1){
+        // total kick
+        double const Fr = hypot(Fx_star, Fy_star) * LocalParticle_get_rpp(part); // rad
+
+        // bending radius is over this distance (half slice length)
+        /*gpuglmem*/ double const dz = .5*BeamBeamBiGaussian3DData_get_slices_other_beam_zeta_bin_width_star(el, i_slice);
+        synrad(part, record, table_index, table, Fr, dz);
```

Comment:
Here the part is passed because I access several properties of it. After each photon emission I do `LocalParticle_add_to_energy(part, energy-initial_energy, 0);` to update the particle energy. This also changes pzeta and delta.

Reply:
Could you do everything with pzeta maybe (computing delta by yourself if needed)

Reply:
see [1 hour ago](https://github.com/xsuite/xfields/pull/50#discussion_r1003523436)